repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -0,0 +1,147 @@
+import json
+from collections import namedtuple
+
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRedHatSignedRPM, RpmTransactionTasks, RepositoriesSetupTasks
+
+
+REPOSITORIES_BLACKLIST = ('rhel8-buildroot', 'rhel8-crb')
+# FIXME: this mapping is not complete and will need to be manually updated frequently
+REPOSITORIES_MAPPING = {
+    'rhel8-appstream': 'rhel-8-for-x86_64-appstream-htb-rpms',
+    'rhel8-baseos': 'rhel-8-for-x86_64-baseos-htb-rpms'}
+
+
+Event = namedtuple('Event', ['action', 'in_pkgs', 'out_pkgs'])
+
+
+def parse_action(action_id):
+    """ Get action label from PES event's action id """
+    labels = ('Present', 'Removed', 'Deprecated', 'Replaced', 'Split', 'Merged', 'Moved', 'Renamed')
+
+    if action_id < 0 or action_id >= len(labels):
```

Comment:
I personally prefer this style of checks:
```python
if -1 < action_id < len(labels):
    return labels[action_id]
raise ValueError('Found event with invalid action ID: {}'. format(action_id))
```
But this will work too